python/lsst/daf/butler/core/storageClass.py

```diff
@@ -181,7 +211,7 @@ def __contains__(self, storageClassOrName):
             True if the supplied string is present, or if the supplied
             `StorageClass` is present and identical.
         """
-        if isinstance(storageClassOrName, "str"):
+        if isinstance(storageClassOrName, str):
```

Comment:
Pre-existing, but do we need the check for "identical" here?  I agree we need to compare for equality in `registerStorageClass`, after first checking whether the name exists, but that only uses `str` code path here.  And having
```
(storageClass in StorageClassFactory()) != (storageClass.name in StorageClassFactory())
```
might be confusing.

Reply:
This raises the question of what "in" means. Does it just mean "is there something with this name already registered" or does it also mean "is this particular storageclass already registered".  It combines the semantics of a `in` for `dict` (the key is checked) and `in` for `list` or `set` (the value is checked) but the caller gets to decide whether to check the storage class or the storage class name.

Reply:
I don't have a terribly strong opinion on this; it just surprised me a bit so I wanted to raise it as a potential issue.